chan_test.go

```diff
@@ -78,43 +78,6 @@ func TestChanResponsePair(t *testing.T) {
 	}
 }
 
-func TestSingle1(t *testing.T) {
```

Comment:
What was racy about this test?

Reply:
I'm not 100% sure myself. The race detector complains that the [`Close` call](https://github.com/ipfs/go-ipfs-cmds/pull/112/files/1428e703db2cf9135d0c1f710fac9391f48716d1..a011832acc5d8d1bd013da9d1422f9f497989ede#diff-28e0ba9683655894160fb3fe3c0c6aceL94) and the [`Next` call](https://github.com/ipfs/go-ipfs-cmds/pull/112/files/1428e703db2cf9135d0c1f710fac9391f48716d1..a011832acc5d8d1bd013da9d1422f9f497989ede#diff-28e0ba9683655894160fb3fe3c0c6aceL101) are unsynchronized writes, but I just don't see it. The test was basically the same as `TestSingleChan`, so I removed this one.

Reply:
Then there is a bug somewhere chanResponse.

Reply:
Ah. I see, the test is racy.